evm/rlp/headers.py

```diff
@@ -70,21 +79,21 @@ class BlockHeader(rlp.Serializable):
     ]
 
     def __init__(self,
-                 difficulty,
-                 block_number,
-                 gas_limit,
-                 timestamp=None,
-                 coinbase=ZERO_ADDRESS,
-                 parent_hash=ZERO_HASH32,
-                 uncles_hash=EMPTY_UNCLE_HASH,
-                 state_root=BLANK_ROOT_HASH,
-                 transaction_root=BLANK_ROOT_HASH,
-                 receipt_root=BLANK_ROOT_HASH,
-                 bloom=0,
-                 gas_used=0,
-                 extra_data=b'',
-                 mix_hash=ZERO_HASH32,
-                 nonce=GENESIS_NONCE):
+                 difficulty: int,
+                 block_number: int,
+                 gas_limit: int,
+                 timestamp: Optional[int] = None,
```

Comment:
I *think* that when `None` is the default value that `Optional[...]` is inferred if you just use `int`.

Reply:
Well, to be precise `mypy` infers that to `Union[builtins.int, builtins.None]` which is basically just a longer version of `Optional[int]`. However, I would not recommend to annotate it as `int` simply because I think the annotation should be aligned with the actual type (e.g. the one mypy would infer itself). So a developer looking at the code can directly see that passing a value is optional here. Your call!

Reply:
I'm going to go with just `int` under the reasoning that the `=None` portion of this is sufficient to signal that the value is option.

Reply:
No problem. I updated all of these cases from `Optional[T]` to just `T`